src/Handlers/DatabaseHandler.php

```diff
@@ -31,6 +38,7 @@ class DatabaseHandler extends ArrayHandler
     public function __construct()
     {
         $this->table = config('Settings')->database['table'] ?? 'settings';
+        $this->group = config('Settings')->database['group'] ?? null;
```

Comment:
Looking at the code I think it actually makes more sense to load up the `BaseConnection` instance here with the proper DBGroup and then replace all the `db_connect()` calls to use the object reference instead.

Reply:
@daycry What do you think of this? I am still interested in it, as I think it reflects the actual dependency more directly. Something like:
```php
    /**
     * The database connection to use.
     */
    private BaseConnection $db;

    public function __construct(?BaseConnection $db = null)
    {
        $this->db = $db ?? db_connect(config('Settings')->database['group']);
    }
```

Reply:
We could probably even hone in on the actual Builder, since it has call through to the connection, and drop the `$group` and `$table` properties altogether:
```php
    /**
     * The Query Builder for the Settings table.
     */
    private Builder $builder;

    public function __construct(?Builder $builder = null)
    {
        $this->builder = $builder ?? db_connect(config('Settings')->database['group'])->table(config('Settings')->database['table']);
    }
```

Reply:
Nice, You avoid duplicating connection lines.

But I don't see any use in passing the parameter in the constructor.

Let me make some modifications.